### PR TITLE
Add rocket.chat

### PR DESCRIPTION
**Pull request by pierreozoux**

### CODE REVIEW

File: stable/rocketchat/templates/deployment.yaml
Comment (geekgonecrazy) [inline stable/rocketchat/templates/deployment.yaml]: To increase consistency with other Rocket.Chat documentation maybe use rocketchat as the db name here?

File: stable/rocketchat/templates/ingress.yaml
Comment (geekgonecrazy) [inline stable/rocketchat/templates/ingress.yaml]: This automatically assumes they have an nginx ingress running.  Is this safe?

Comment (linki) [reply stable/rocketchat/templates/ingress.yaml]: It should be commented out by default. Similar to https://github.com/kubernetes/charts/blob/e80d1cfb8857cc71096b3b8a4638a789b378db88/stable/grafana/values.yaml#L23

This will allow the Chart to work by default. People that use the nginx-based ingress controller can either opt-in by setting this to "nginx" or they configure their cluster/ingress-controller to pick up Ingresses without annotations, too (like GKE does).

**Comment by linki** [comment]
@pierreozoux can you have another look at this?

@geekgonecrazy Thanks for your thoughts.

**Comment by pierreozoux** [comment]
@linki I addressed all the comments already (I didn't comment, I just fixed it to avoid noise).

But yeah, thanks @geekgonecrazy for your comments, I fixed them all :)

Let me know if there is anything more needed?

File: stable/rocketchat/.helmignore
Comment (linki) [inline stable/rocketchat/.helmignore:1]: This is not the standard `.gitignore`. Let's wait on the outcome of https://github.com/kubernetes/charts/pull/485/files#r106305261

Looks like we prefer the "canonical" helmignore from now on but many charts still have just this. It probably makes more sense to sync them up in a separate PR across all charts.

**Comment by pierreozoux** [comment]
Is there anything I can do to get this merged?

**Comment by engelgabriel** [comment]
The Rocket.Chat team is looking forward for this!

**Comment by geekgonecrazy** [comment]
@pierreozoux one thing to maybe add here is the environment variables to populate the INSTANCE_IP variable with the pod ip.

Here is a snippet I have for adding this to the schema:
```
name: INSTANCE_IP
valueFrom:
  fieldRef:
    fieldPath: status.podIP
```

This way if scaled the instances can know how to talk to each other over the overlay network

**Comment by pierreozoux** [comment]
@geekgonecrazy hum, do you have the link to the documentation?

If the person uses S3, it would scale automatically out of the box.
Let's think about this INSTANCE_IP (which is a better way I guess) for another PR, once this got merged?

**Comment by geekgonecrazy** [comment]
@pierreozoux 
Here's a snippet directly out of one of our deployments:
testdeploy.yaml
<details>

```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  labels:
    DeploymentDomain: testdeploy.rocket.chat
    DeploymentName: testdeploy
  name: testdeploy
  namespace: <namespace here>
spec:
  replicas: 2
  revisionHistoryLimit: 1
  selector:
    matchLabels:
      DeploymentDomain: testdeploy.rocket.chat
      DeploymentName: testdeploy
  strategy:
    rollingUpdate:
      maxSurge: 1
      maxUnavailable: 1
    type: RollingUpdate
  template:
    metadata:
      creationTimestamp: null
      labels:
        DeploymentDomain: testdeploy.rocket.chat
        DeploymentName: testdeploy
    spec:
      containers:
      - env:
        - name: INSTANCE_IP
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: status.podIP
        - <another 10-20 ommited for briefness>
        image: rocketchat/rocket.chat:0.55.1
        imagePullPolicy: IfNotPresent
        name: testdeploy
        ports:
        - containerPort: 3000
          protocol: TCP
```

</details>
Its for sure something that could be added in later.

**Comment by viglesiasce** [comment]
@k8s-bot ok to test

File: stable/rocketchat/templates/deployment.yaml
Comment (viglesiasce) [inline stable/rocketchat/templates/deployment.yaml]: What is this line trying to do? Shouldnt it be:

{{- if .Values.host -}}

Comment (viglesiasce) [reply stable/rocketchat/templates/deployment.yaml]: The linter is also complaining about it and I the pod is crashing with the following when it is not set:

```
/app/bundle/programs/server/boot.js:356
}).run();
   ^

Error: $ROOT_URL, if specified, must be an URL
    at Error (native)
    at packages/meteor.js:1218:13
    at packages/meteor.js:1233:4
    at packages/meteor.js:1380:3
    at /app/bundle/programs/server/boot.js:303:34
    at Array.forEach (native)
    at Function._.each._.forEach (/app/bundle/programs/server/node_modules/underscore/underscore.js:79:11)
    at /app/bundle/programs/server/boot.js:128:5
    at /app/bundle/programs/server/boot.js:352:5
    at Function.run (/app/bundle/programs/server/profile.js:510:12)
```

If this is a required field look at what ghost and gitlab-ce have done for this use case.

**Review by viglesiasce — CHANGES_REQUESTED** [review]

**Comment by geekgonecrazy** [comment]
@pierreozoux do you need any help getting these changes made?  Would be glad to help out

**Comment by pierreozoux** [comment]
@viglesiasce thanks I addressed your comment.

**Comment by pierreozoux** [comment]
Is there anything else I can do to help?

**Comment by geekgonecrazy** [comment]
@viglesiasce / @linki we would love to see this progress forward.  Is there anything you guys need on this to get it merged?

File: stable/rocketchat/Chart.yaml
Comment (unguiculus) [inline stable/rocketchat/Chart.yaml]: Use Github username.

File: stable/rocketchat/templates/deployment.yaml
Comment (unguiculus) [inline stable/rocketchat/templates/deployment.yaml]: `app` label should be `{{ template "name" . }}`. Apply everywhere.

File: stable/rocketchat/templates/svc.yaml
Comment (unguiculus) [inline stable/rocketchat/templates/svc.yaml]: After changing the `app` label as mentioned above, the selector should be as follows:

```yaml
app: {{ template "name" . }}
release: "{{ .Release.Name }}"
```

File: stable/rocketchat/values.yaml
Comment (unguiculus) [inline stable/rocketchat/values.yaml]: We've started recommending not to specify default resources.

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by pierreozoux** [comment]
@unguiculus Thanks for your comment, I addressed them.

File: stable/rocketchat/Chart.yaml
Comment (unguiculus) [inline stable/rocketchat/Chart.yaml:18]: Name should be a Github user.

File: stable/rocketchat/templates/NOTES.txt
Comment (unguiculus) [inline stable/rocketchat/templates/NOTES.txt]: Fix label selectors:
`-l app="{{ template "fullname" . }},release={{ .Release.Name }}"`

File: stable/rocketchat/templates/deployment.yaml
Comment (unguiculus) [inline stable/rocketchat/templates/deployment.yaml]: This is empty by default. When I install the chart with defaults only MongoDB comes up. It should be possible to install a chart with default values.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/rocketchat/Chart.yaml
Comment (geekgonecrazy) [reply stable/rocketchat/Chart.yaml:18]: @unguiculus this is the org name.  Do we need to add individual maintainers listed here?

Comment (geekgonecrazy) [reply stable/rocketchat/Chart.yaml:18]: @pierreozoux I can provide a list from Rocket.Chat that would handle any issues that would arise as well as of course your self.

Comment (geekgonecrazy) [reply stable/rocketchat/Chart.yaml:18]: @unguiculus I notice many of the bitnami created ones refer to their org name on github and add an email address.  But i've gone ahead and added two additional maintainers by github username.

File: stable/rocketchat/templates/NOTES.txt
Comment (geekgonecrazy) [reply stable/rocketchat/templates/NOTES.txt]: Fixed, also added these labels to the actual pod level so the command would work in the first place :)

File: stable/rocketchat/templates/deployment.yaml
Comment (geekgonecrazy) [reply stable/rocketchat/templates/deployment.yaml]: Fixed

**Comment by geekgonecrazy** [comment]
Bump to verify CLA

**Comment by geekgonecrazy** [comment]
@unguiculus @viglesiasce addressed feedback. 

* Added maintainers.  I went ahead and left our org and email for contact, and added two maintainers to the list.
* Fixed Notes so the output will suggest how to get site going at http://chat.yourdomain.com.  Fixed a few selectors in the queries it says to run as well as the port forward port
* Now can be installed with as little as: `helm install stable/rocketchat` brings up mongodb and rocket.chat

I've also signed the CLA

**Comment by geekgonecrazy** [comment]
Helping @pierreozoux get things finished here.

@unguiculus @viglesiasce I think i've now addressed all feedback. 

* Added maintainers.  I went ahead and left our org and email for contact, and added two maintainers to the list.
* Fixed Notes so the output will suggest how to get site going at http://chat.yourdomain.com.  Fixed a few selectors in the queries it says to run as well as the port forward port
* Now can be installed with as little as: `helm install stable/rocketchat` brings up mongodb and rocket.chat

I've also signed the CLA

**Comment by geekgonecrazy** [comment]
@viglesiasce @unguiculus @linki would any of you guys be able to take a look at this?

**Comment by viglesiasce** [comment]
@geekgonecrazy I'll take a look today.

**Comment by geekgonecrazy** [comment]
@viglesiasce thank you!  I've also joined in the slack #helm-users channel and glad to chat and address what ever concerns if any still remain.

**Review by viglesiasce — APPROVED** [review]

**Comment by viglesiasce** [comment]
Worked a treat! Thanks!

**Comment by geekgonecrazy** [comment]
Perfect!  Thanks @viglesiasce !

**Comment by pierreozoux** [comment]
\o/ Thanks!

**Comment by sathieu** [comment]
Hello,

It is not on https://kubeapps.com/. Can someone push it?

**Comment by sathieu** [comment]
and there is no README...

**Comment by unguiculus** [comment]
@prydonius Could you check why the chart is not listed?

**Comment by geekgonecrazy** [comment]
Would be glad to help make what ever changes are needed to make it show up.

**Comment by sathieu** [comment]
Maybe this is the README missing?

**Comment by beranPro** [comment]
Hi @pierreozoux @viglesiasce ,
rocket.chat is still not available at kubeapps.
Other projects on helm have Readme.md's and they are listet on kubeapps and the readme is displayed.
Have anybode more information? How can we fix this?

Thanks a lot!

**Comment by geekgonecrazy** [comment]
@beranPro @sathieu is it just a matter of missing readme?  If you could point me to what ever docs related to this i'll gladly put together

**Comment by shinebayar-g** [comment]
Please maintain your helm chart up to date.

**Comment by geekgonecrazy** [comment]
we are aware of an issue created by an update to the helm chart here #7636

The updated mongo dependency introduced an automatic password generation that wasn't present before.  Passing: `--set mongodb.usePassword=false` helm install temporarily solves, we have a fix incoming soon.  

cc: @LuluGO

**Pull request merged**
